# HG changeset patch
# User Rik <rik@octave.org>
# Date 1398807465 25200
#      Tue Apr 29 14:37:45 2014 -0700
# Node ID ea22cf3e370b74ba60ddae34edff5aa38dd19dbf
# Parent  abd8d8c686c942c9e14c327ad5d0cce55c9bf984
Don't plot grid lines directly over axis box lines.

* gl-render.cc (opengl_renderer::render_grid): Only plot grid lines which are
within the axes box rather than within or on the axes box boundary.

diff --git a/libinterp/corefcn/gl-render.cc b/libinterp/corefcn/gl-render.cc
--- a/libinterp/corefcn/gl-render.cc
+++ b/libinterp/corefcn/gl-render.cc
@@ -646,17 +646,17 @@ opengl_renderer::render_grid (const std:
                               double p1, double p1N, double p2, double p2N,
                               int xyz, bool is_3D)
 {
   set_linestyle (gridstyle, true);
   glBegin (GL_LINES);
   for (int i = 0; i < ticks.numel (); i++)
     {
       double val = ticks(i);
-      if (lim1 <= val && val <= lim2)
+      if (lim1 < val && val < lim2)
         {
           if (xyz == 0) // X
             {
               glVertex3d (val, p1N, p2);
               glVertex3d (val, p1, p2);
               if (is_3D)
                 {
                   glVertex3d (val, p1, p2N);
